tests/bluetooth/bsim_bt/bsim_test_l2cap/src/main_l2cap_ecred.c

```diff
@@ -0,0 +1,608 @@
+{
+	for (int i = 0; i < ARRAY_SIZE(channels); i++) {
+		if (channels[i].in_use) {
+			LOG_DBG("Disconnecting channel: %d)", channels[i].chan_id);
+			int err = bt_l2cap_chan_disconnect(&channels[i].le.chan);
+
+			if (err) {
+				LOG_DBG("can't disconnnect channel (err: %d)", err);
+			}
+
+			channels[i].in_use = false;
+		}
+	}
+}
+
+static int accept(struct bt_conn *conn, struct bt_l2cap_chan **l2cap_chan)
+{
+	struct channel *chan;
+
+	chan = get_free_channel();
+	if (!chan) {
+
+		return -ENOMEM;
+	}
+
+	*l2cap_chan = &chan->le.chan;
+
+	return 0;
+}
+
+static struct bt_l2cap_server *get_free_server(void)
+{
+	for (int i = 0; i < SERVERS; i++) {
+		if (servers[i].psm) {
+			continue;
+		}
+
+		return &servers[i];
+	}
+
+	return NULL;
+}
+
+static void register_l2cap_server(void)
+{
+	struct bt_l2cap_server *server;
+
+	server = get_free_server();
+	if (!server) {
+		FAIL("Failed to get free server");
+		return;
+	}
+
+	server->accept = accept;
+	server->psm = 0;
+
+	if (bt_l2cap_server_register(server) < 0) {
+		FAIL("Failed to get free server");
+
+		return;
+	}
+
+	LOG_DBG("L2CAP server registered, PSM:0x%X", server->psm);
+}
+
+static void connected(struct bt_conn *conn, uint8_t conn_err)
+{
+	char addr[BT_ADDR_LE_STR_LEN];
+
+	bt_addr_le_to_str(bt_conn_get_dst(conn), addr, sizeof(addr));
+
+	if (conn_err) {
+		FAIL("Failed to connect to %s (%u)", addr, conn_err);
+		bt_conn_unref(default_conn);
+		default_conn = NULL;
+
+		return;
+	}
+
+	default_conn = bt_conn_ref(conn);
+	LOG_DBG("%s", addr);
+
+	SET_FLAG(is_connected);
+}
+
+static void disconnected(struct bt_conn *conn, uint8_t reason)
+{
+	char addr[BT_ADDR_LE_STR_LEN];
+
+	bt_addr_le_to_str(bt_conn_get_dst(conn), addr, sizeof(addr));
+
+	LOG_DBG("%s (reason 0x%02x)", addr, reason);
+
+	if (default_conn != conn) {
+		FAIL("Conn mismatch disconnect %s %s)", default_conn, conn);
+
+		return;
+	}
+
+	bt_conn_unref(default_conn);
+	default_conn = NULL;
+	UNSET_FLAG(is_connected);
+}
+
+BT_CONN_CB_DEFINE(conn_callbacks) = {
+	.connected = connected,
+	.disconnected = disconnected,
+};
+
+static void send_sdu(int iteration, int chan_idx, int bytes)
+{
+	struct bt_l2cap_chan *chan = &channels[chan_idx].le.chan;
+	struct net_buf *buf;
+
+	/* First 4 bytes in sent payload is iteration count */
+	*((int *)channels[chan_idx].payload) = iteration;
+
+	if (channels[chan_idx].buf != 0) {
+		FAIL("Buf should have been dealocated by now");
+	}
+
+	if (chan_idx == 0) {
+		buf = net_buf_alloc(&tx_data_pool_0, K_NO_WAIT);
+	} else {
+		buf = net_buf_alloc(&tx_data_pool_1, K_NO_WAIT);
+	}
+
+	if (buf == NULL) {
+		FAIL("Failed to get buff on ch %i, iteration %i should never happen", chan_idx,
+		     chan_idx);
+	}
+
+	channels[chan_idx].buf = buf;
+	net_buf_reserve(buf, BT_L2CAP_CHAN_SEND_RESERVE);
+	net_buf_add_mem(buf, channels[chan_idx].payload, bytes);
+
+	LOG_DBG("bt_l2cap_chan_sending ch: %i bytes: %i iteration: %i", chan_idx, bytes, iteration);
+	int ret = bt_l2cap_chan_send(chan, buf);
```

Comment:
```suggestion
	const int ret = bt_l2cap_chan_send(chan, buf);
```

Reply:
This comment basically applies to all occurrences of variables that can be defined as `const` :D

Reply:
would be much better if this kind of ambiguity would be checked be a script, should be easy to identify the vars which were written only once, so to avoid the manual work

Reply:
That should indeed be possible :)